utils/field.js

```diff
@@ -453,6 +408,63 @@ function getEnumValuesWithCustomValues(enumValues) {
   });
 }
 
+function prepareClientFieldForTemplates(entityWithConfig, field, generator) {
+  if (field.fieldValidateRulesPatternAngular === undefined) {
+    field.fieldValidateRulesPatternAngular = field.fieldValidateRulesPattern
+      ? field.fieldValidateRulesPattern.replace(/"/g, '&#34;')
+      : field.fieldValidateRulesPattern;
+  }
+
+  if (field.fieldValidateRulesPatternReact === undefined) {
+    field.fieldValidateRulesPatternReact = field.fieldValidateRulesPattern
+      ? field.fieldValidateRulesPattern.replace(/'/g, "\\'")
+      : field.fieldValidateRulesPattern;
+  }
+}
+
+function prepareServerFieldForTemplates(entityWithConfig, field, generator) {
+  if (field.fieldNameAsDatabaseColumn === undefined) {
+    const fieldNameUnderscored = _.snakeCase(field.fieldName);
+    const jhiFieldNamePrefix = generator.getColumnName(entityWithConfig.jhiPrefix);
+
+    if (isReservedTableName(fieldNameUnderscored, entityWithConfig.prodDatabaseType)) {
```

Comment:
Ideally a specific method by check (could be reused by blueprints I guess)

Reply:
jdl validation can be improved for modularity.

Reply:
We should think about this for v8.